components/src/core/components/PopOver/__tests__/pop-over.spec.ts

```diff
@@ -83,6 +84,190 @@ describe('PopOver', () => {
     expect(wrapper.find('.oxd-pop-over-content').exists()).toBeFalsy();
   });
 
+  it('click trigger button again to close the content when persistent is set to true', async () => {
+    const wrapper = mount(PopOver, {
+      slots: {
+        button: '<oxd-icon-button class="popover-icon" name="funnel" />',
+        default: '<div>This is PopOver Content</div>',
+      },
+      props: {
+        persistent: true,
+      },
+      global: {
+        stubs: {
+          'oxd-icon-button': {
+            template: '<i />',
+          },
+        },
+      },
+    });
+
+    const button = wrapper.find('.popover-icon');
+
+    expect(wrapper.find('.oxd-pop-over-button').exists()).toBeTruthy();
+    expect(wrapper.find('.oxd-pop-over-content').exists()).toBeFalsy();
+
+    await button.trigger('click');
+    await wrapper.vm.$nextTick();
+    expect(wrapper.find('.oxd-pop-over-button').exists()).toBeTruthy();
+    expect(wrapper.find('.oxd-pop-over-content').exists()).toBeTruthy();
+
+    await button.trigger('click');
+    await wrapper.vm.$nextTick();
+    expect(wrapper.find('.oxd-pop-over-button').exists()).toBeTruthy();
+    expect(wrapper.find('.oxd-pop-over-content').exists()).toBeFalsy();
+  });
+
+  it('click outside pop over to close the content when persistent is set to false', async () => {
+    const wrapper = mount(PopOver, {
+      slots: {
+        button: '<oxd-icon-button class="popover-icon" name="funnel" />',
+        default: '<div>This is PopOver Content</div>',
+      },
+      props: {
+        persistent: false,
+      },
+      global: {
+        stubs: {
+          'oxd-icon-button': {
+            template: '<i />',
+          },
+        },
+      },
+    });
+
+    const outsideElement = document.createElement('button');
+    outsideElement.classList.add('outside-element');
+    document.body.appendChild(outsideElement);
+
+    try {
```

Comment:
I think assertions should not be made inside try/catch because jest will not fail the test if the code goes to catch.

https://github.com/jestjs/jest/issues/3917

Reply:
removed